### PR TITLE
Feature/rb 92 function receive enemy death

### PR DESCRIPTION
**Pull request by guillaumeAbel**
* **Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [x] Norm of the code has been respected

* **In what state is your pull request?**
- [x] Ready to merge / Waiting a reviwer to see my work.
- [ ] Work In Progress (WIP) / My work is not finish but i want daily reviews. (Draft)
- [ ] CI Review / I want to see what the CI thinks of my work.
- [ ] Need feedback / Just to have feedback on what i produced. (May not be merged)

* **What kind of change does this PR introduce?** (You can choose multiple)
- [ ] Bug fix
- [x] Feature request
- [ ] New / Updated documentation
- [ ] Testing CI ( Make the pull request in draft mode)

* **What is the current behavior?** (link an issue based on the kind of change this pr introduce)

* **What is the new behavior (if this is a feature change)?**

* **Other information**:

### CODE REVIEW

File: src/ECS/ECSCustomTypes.hpp [hunk 1]
```diff
@@ -65,7 +64,16 @@ namespace Types {
...
-    struct Enemy { };
+    static long int enemyNb = 0;
+
+    struct Enemy {
+            Enemy()
+            {
+                constId.value = enemyNb;
+                enemyNb++;
+            }
+            struct enemy_id_s constId;
```
Comment (romainpanno) [inline src/ECS/ECSCustomTypes.hpp]: make it const

File: src/ECS/MessageTypes.h [hunk 2]
```diff
@@ -13,4 +13,8 @@ PACK(struct health_s {
     int hp;
 });
 
+PACK(struct enemy_id_s {
+    long int value;
```
Comment (romainpanno) [inline src/ECS/MessageTypes.h]: make it unsigned int

File: src/ECS/ECSCustomTypes.hpp [hunk 3]
```diff
@@ -65,7 +64,16 @@ namespace Types {
 
     struct EnemyAllies { };
 
-    struct Enemy { };
+    static long int enemyNb = 0;
+
+    struct Enemy {
+            Enemy()
```
Comment (romainpanno) [inline src/ECS/ECSCustomTypes.hpp]: change it by taking the id instead of attributing it yourself

**Review by romainpanno — CHANGES_REQUESTED** [review]

File: src/ECS/MessageTypes.h (hunk 2 above)
Comment (Saverio976) [inline src/ECS/MessageTypes.h]: long int ?

genre y auras plus de 2147483647 ennemy ?

File: src/ECS/ECSCustomTypes.hpp [hunk 4]
```diff
@@ -65,7 +64,16 @@ namespace Types {
 
     struct EnemyAllies { };
 
-    struct Enemy { };
+    static long int enemyNb = 0;
```
Comment (Saverio976) [inline src/ECS/ECSCustomTypes.hpp]: bonsoir non

Comment (Saverio976) [reply src/ECS/ECSCustomTypes.hpp]: mais la en static private dans la class bg

**Review by Saverio976 — CHANGES_REQUESTED** [review]

File: src/ECS/MessageTypes.h (hunk 2 above)
Comment (romainpanno) [reply src/ECS/MessageTypes.h]: it will be unsigned int, is it ok ?, its same length of an int

File: src/ECS/ECSCustomTypes.hpp [hunk 5]
```diff
@@ -10,12 +10,11 @@
 #include <cstddef>
 #include <functional>
 #include <optional>
+#include "MessageTypes.h"
```
Comment (Saverio976) [inline src/ECS/ECSCustomTypes.hpp]:
```suggestion
external "C" {
#include "MessageTypes.h"
}
```

**Review by Saverio976 — DISMISSED** [review]

**Comment by guillaumeAbel** [comment]
actually not compiling, cause the bug @TTENSHII is actually fixing, Enemy component is copied, but it cannot be cause it have a const value

**Review by kw4ntiq — APPROVED** [review]

File: src/ECS/MessageTypes.h [hunk 6]
```diff
@@ -13,4 +13,8 @@ PACK(struct health_s {
     int hp;
 });
 
+PACK(struct enemy_id_s {
+    unsigned int value;
```
Comment (Saverio976) [inline src/ECS/MessageTypes.h]: value ? c'est quoi

**Review by romainpanno — APPROVED** [review]

**Review by romainpanno — APPROVED** [review]

**Pull request merged**